lib/graphql_schema.rb

```diff
@@ -2,6 +2,24 @@
 require 'set'
 
 class GraphQLSchema
+  INVALID_SCHEMA = Class.new(StandardError)
+
+  class << self
+    def load_schema(schema)
+      case schema
+      when Hash
+        new(schema)
+      when GraphQLSchema
+        schema
+      when Pathname, String
+        schema = File.read(schema) if Pathname(schema).file?
```

Comment:
If you have an instance of `GraphQLSchema` then there is no need to call this method.  If you have a schema hash, then you can just use `GraphQLSchema.new(schema_hash)`.  I don't see the need for a single method that handles all these cases, since it causes confusion about how a string should be treated.

Another possible interpretation of a string would be a url, which we might want to also support in the future.